src/eset.c

```diff
@@ -155,6 +156,77 @@ eset_remove(eset_t *eset, edata_t *edata) {
 	    cur_extents_npages - (size >> LG_PAGE), ATOMIC_RELAXED);
 }
 
+edata_t *
+eset_enumerate_alignment_search(eset_t *eset, size_t size, pszind_t bin_ind,
+    size_t alignment) {
+	if (edata_heap_empty(&eset->bins[bin_ind].heap)) {
+		return NULL;
+	}
+
+	void *bfs_queue[ESET_ENUMERATE_NUM];
+	uint16_t front, rear;
+	size_t queue_size, visited_num;
+	edata_t *edata = NULL;
+	edata_heap_enumerate_prepare(&eset->bins[bin_ind].heap, bfs_queue,
+	    &front, &rear, &queue_size, &visited_num, ESET_ENUMERATE_NUM);
+	while ((edata =
+	    edata_heap_enumerate_next(&eset->bins[bin_ind].heap, bfs_queue,
+	    &front, &rear, &queue_size, &visited_num, ESET_ENUMERATE_NUM,
+	    ESET_ENUMERATE_NUM))) {
+		uintptr_t base = (uintptr_t)edata_base_get(edata);
+		size_t candidate_size = edata_size_get(edata);
+		if (candidate_size < size) {
+			continue;
+		}
+
+		uintptr_t next_align = ALIGNMENT_CEILING((uintptr_t)base,
+		    PAGE_CEILING(alignment));
```

Comment:
Is PAGE_CEILING(alignment) necessary? Since this is already at the eset level. Maybe turn it into asserts